src/vm/gccover.cpp

```diff
@@ -43,25 +43,34 @@ static void replaceSafePointInstructionWithGcStressInstr(UINT32 safePointOffset,
 static bool replaceInterruptibleRangesWithGcStressInstr (UINT32 startOffset, UINT32 stopOffset, LPVOID codeStart);
 #endif
 
+// There is a call target instruction, try to find the MethodDesc for where target points to.
+// Returns nullptr if it can't find it.
 static MethodDesc* getTargetMethodDesc(PCODE target)
 {
     MethodDesc* targetMD = ExecutionManager::GetCodeMethodDesc(target);
-    if (targetMD == 0) 
+    if (targetMD != nullptr)
     {
-        VirtualCallStubManager::StubKind vsdStubKind = VirtualCallStubManager::SK_UNKNOWN;
-        VirtualCallStubManager *pVSDStubManager = VirtualCallStubManager::FindStubManager(target, &vsdStubKind);
-        if (vsdStubKind != VirtualCallStubManager::SK_BREAKPOINT && vsdStubKind != VirtualCallStubManager::SK_UNKNOWN)
-        {
-            DispatchToken token = VirtualCallStubManager::GetTokenFromStubQuick(pVSDStubManager, target, vsdStubKind);
-            _ASSERTE(token.IsValid());
-            targetMD = VirtualCallStubManager::GetInterfaceMethodDescFromToken(token);
-        }
-        else
-        {
-            targetMD = AsMethodDesc(size_t(MethodDesc::GetMethodDescFromStubAddr(target, TRUE)));
-        }
+        // It is JIT/NGened call.
+        return targetMD;
+    }
+
+    VirtualCallStubManager::StubKind vsdStubKind = VirtualCallStubManager::SK_UNKNOWN;
+    VirtualCallStubManager *pVSDStubManager = VirtualCallStubManager::FindStubManager(target, &vsdStubKind);
+    if (vsdStubKind != VirtualCallStubManager::SK_BREAKPOINT && vsdStubKind != VirtualCallStubManager::SK_UNKNOWN)
+    {
+        // It is a VSD stub manager.
+        DispatchToken token = VirtualCallStubManager::GetTokenFromStubQuick(pVSDStubManager, target, vsdStubKind);
+        _ASSERTE(token.IsValid());
+        return VirtualCallStubManager::GetInterfaceMethodDescFromToken(token);
+    }
+
+    if (RangeSectionStubManager::GetStubKind(target) != STUB_CODE_BLOCK_UNKNOWN)
+    {
+        // The address looks like a value stub, try to get the method descriptor.
+        return AsMethodDesc(size_t(MethodDesc::GetMethodDescFromStubAddr(target, TRUE)));
```

Comment:
We cast `MethodDesc*` to `size_t` and then call `AsMethodDesc` that returns the original address or `nullptr` if the address did not pass checks. Looks like it was done to strip cases where `GetMethodDescFromStubAddr` returned `non-nullptr address` for bad targets. 

Do we still need this check after adding `if (RangeSectionStubManager::GetStubKind(target) != STUB_CODE_BLOCK_UNKNOWN)` or it can be deleted?

Reply:
I think you can delete if you change the check to:

`if (RangeSectionStubManager::GetStubKind(target) == STUB_CODE_BLOCK_PRECODE)`

There should not be any false positives that get past this that need to be filtered out.

Reply:
Thank you.